### ISSUE TITLE
Including libraries for the Node codegen

### ISSUE THREAD

**Issue by justjoheinz** [idris-lang/Idris-dev#3697]
Disclaimer: Maybe my thinking is wrong here

I wanted to experiment a bit with the node codegen. In order to write a wrapper library I naturally install modules using `npm install xyz`.
The resulting module will be located in e.g. `node_modules/xyz/main.js`

If I include this using:

    %lib Node "xyz/main"

the generated javascript file will include the line 

    var xyz/main = require("xyz/main")

The code generator should probably either just take the last section of the require path or provide an optional parameter like 

    $lib Node "xyz/main"  as "client"

to produce

    var client = require("xyz/main")

Another related question - how would I tell idris to look inside of node_modules for the required files? Or is this done by `node` itself when executing the javascript?

**Comment by melted** [comment]
Sounds like a good idea. 

Node connects your code to the libs from whats in package.json. So that's not a problem.

**Comment by justjoheinz** [comment]
I have been digging in the code a bit, unfortunately I never managed to get idris to compile locally and I have no big haskell experience, so I cannot test the code.

But modifying `CodegenJavaScript.hs` like this:

```haskell
      includeLibs :: [String] -> String
      includeLibs =
        let
          repl '\' = "_"
          repl '/' = "_"
          repl c   = c
        in
          concatMap (\lib -> "var " ++ (repl <$> lib) ++ " = require(\"" ++ lib ++"\");\n")
```
should a least output identifiers for node packages, which are sound. It does not cover the idea to have an alias, but would be a step forward.

**Comment by rbarreiro** [comment]
Implemented as suggested by #3855

**Comment by ahmadsalim** [comment]
I will close this since the PR was accepted.
